shell/platform/darwin/common/framework/Source/FlutterChannels.mm

```diff
@@ -9,12 +9,28 @@
 #pragma mark - Basic message channel
 
 static NSString* const kFlutterChannelBuffersChannel = @"dev.flutter/channel-buffers";
+static NSString* const kResizeMethod = @"resize";
+static NSString* const kOverflowMethod = @"overflow";
 
 static void ResizeChannelBuffer(NSObject<FlutterBinaryMessenger>* binaryMessenger,
                                 NSString* channel,
                                 NSInteger newSize) {
-  NSString* messageString = [NSString stringWithFormat:@"resize\r%@\r%@", channel, @(newSize)];
-  NSData* message = [messageString dataUsingEncoding:NSUTF8StringEncoding];
+  NSArray* args = @[ channel, [NSNumber numberWithInt:newSize] ];
+  FlutterMethodCall* resizeMethodCall = [FlutterMethodCall methodCallWithMethodName:kResizeMethod
+                                                                          arguments:args];
+  NSObject<FlutterMethodCodec>* codec = [FlutterStandardMethodCodec sharedInstance];
+  NSData* message = [codec encodeMethodCall:resizeMethodCall];
+  [binaryMessenger sendOnChannel:kFlutterChannelBuffersChannel message:message];
+}
+
+static void SetAllowChannelOverflow(NSObject<FlutterBinaryMessenger>* binaryMessenger,
```

Comment:
All method and function declarations need declaration comments, per the style guide.